components/tidb_query/src/rpn_expr/impl_cast.rs

```diff
@@ -323,6 +348,140 @@ fn cast_json_as_uint(ctx: &mut EvalContext, val: &Option<Json>) -> Result<Option
     }
 }
 
+// cast any as real, some cast functions reuse `cast_any_as_any`
+//
+// cast_decimal_as_signed_real -> cast_any_as_any<Decimal, Real>
+// cast_time_as_real -> cast_any_as_any<Time, Real>
+// cast_duration_as_real -> cast_any_as_any<Duration, Real>
+// cast_json_as_real -> by cast_any_as_any<Json, Real>
+
+#[rpn_fn]
+#[inline]
+fn cast_signed_int_as_signed_real(val: &Option<Int>) -> Result<Option<Real>> {
+    match val {
+        None => Ok(None),
+        Some(val) => Ok(Real::new(*val as f64).ok()),
+    }
+}
+
+#[rpn_fn(capture = [extra])]
+#[inline]
+fn cast_signed_int_as_unsigned_real(
+    extra: &RpnFnCallExtra,
+    val: &Option<Int>,
+) -> Result<Option<Real>> {
+    match val {
+        None => Ok(None),
+        Some(val) => {
+            if in_union(extra.implicit_args) && *val < 0 {
+                Ok(Some(Real::zero()))
+            } else {
+                // FIXME: TiDB here may has bug, fix this after fix TiDB's
```

Comment:
Is there any issue? Prefer removing it if it is correct in `tidb_query`.

Reply:
negative number to unsigned real's logic may be wrong here.